### PR TITLE
add ExpiresAt field to MigToken status

### PR DESCRIPTION
**Pull request by alaypatel07**
This adds time expiresIn status field which contains UTC 
timestamp at which the token expires

### CODE REVIEW

**Comment by alaypatel07** [comment]
manually tried this

```
[alpatel@alpatel v4]$ oc get migtokens -n bob-destination   -oyaml | grep expires
    expiresIn: "2020-06-03T02:14:11Z"
    expiresIn: "2020-06-02T16:45:20Z"
[alpatel@alpatel v4]$ oc get migtokens -n bob-destination   -oyaml | grep expires -C 2
      status: "True"
      type: Ready
    expiresIn: "2020-06-03T02:14:11Z"
    observedDigest: 72736978276417e7efad0d5d0ae2843fdf841ae79296d6d2feb1e72b5acf365f
- apiVersion: migration.openshift.io/v1alpha1
--
      status: "True"
      type: Ready
    expiresIn: "2020-06-02T16:45:20Z"
    observedDigest: f028e13c06ae9b923898166cf656edb06a5c25c88c5b06826d04b0b4755719ee
kind: List
[alpatel@alpatel v4]
```

File: pkg/apis/migration/v1alpha1/migtoken_types.go [hunk 1]
```diff
@@ -24,7 +26,8 @@ type MigTokenSpec struct {
 // MigTokenStatus defines the observed state of MigToken
 type MigTokenStatus struct {
 	Conditions
-	ObservedDigest string `json:"observedDigest,omitempty"`
+	ExpiresIn      time.Time `json:"expiresIn,omitempty"`
```
Comment (alaypatel07) [inline pkg/apis/migration/v1alpha1/migtoken_types.go:29]: should we name this `ExpiresAt`? I tried to keep it consistent with oauthaccesstoken API but it is in seconds and this is in UTC so doesn't really make sense

Comment (alaypatel07) [reply pkg/apis/migration/v1alpha1/migtoken_types.go]: updated the name so it reads better :)

**Comment by alaypatel07** [comment]
woooot, the change to pointer did it with the builds. This PR is ready to be reviewed

**Comment by alaypatel07** [comment]
```
[alpatel@alpatel v4]$ oc get migtokens -n bob-destination   -oyaml | grep expires -C 2
      status: "True"
      type: Ready
    expiresAt: "2020-06-03T02:14:11Z"
    observedDigest: 72736978276417e7efad0d5d0ae2843fdf841ae79296d6d2feb1e72b5acf365f
- apiVersion: migration.openshift.io/v1alpha1
--
      status: "True"
      type: Ready
    expiresAt: "2020-06-02T16:45:20Z"
    observedDigest: f028e13c06ae9b923898166cf656edb06a5c25c88c5b06826d04b0b4755719ee
kind: List
[alpatel@alpatel v4]$
```
Tried the latest build, works pretty well

**Comment by djwhatle** [comment]
Non admin tracking issue: #388

**Comment by alaypatel07** [comment]
closing in favour of #554

**Pull request closed without merging**